pkg/storage/scrub.go

```diff
@@ -279,21 +279,28 @@ func CheckLayers(
 	}
 
 	for _, layer := range man.Layers {
-		layerContent, err := imgStore.GetBlobContent(imageName, layer.Digest)
+		layerReader, _, err := imgStore.GetBlob(imageName, layer.Digest, "")
 		if err != nil {
 			imageRes = getResult(imageName, tagName, layer.Digest, err)
 
 			break
 		}
 
-		computedDigest := godigest.FromBytes(layerContent)
+		computedDigest, err := godigest.FromReader(layerReader)
+		if err != nil {
+			imageRes = getResult(imageName, tagName, layer.Digest, err)
+
+			break
+		}
 
 		if computedDigest != layer.Digest {
 			imageRes = getResult(imageName, tagName, layer.Digest, errors.ErrBadBlobDigest)
 
 			break
 		}
 
+		_ = layerReader.Close()
```

Comment:
We need to make sure this is called, even on the `break` cases above.
Why not move the entire code block to a separate function?